src/lib/util/unicode.h

```diff
@@ -100,6 +100,10 @@ std::string utf8_from_uchar(char32_t uchar);
 int utf16_from_uchar(char16_t *utf16string, size_t count, char32_t uchar);
 int utf16f_from_uchar(char16_t *utf16string, size_t count, char32_t uchar);
 
+// converting UTF-8 strings to/from "wide" strings
+std::wstring wstring_from_utf8(const std::string &utf8string);
+std::string utf8_from_wstring(const std::wstring &string);
+
```

Comment:
Now that you have this thing, could you also change the code in `src/frontend/mame/ui/filesel.cpp` or wherever it is to use it instead of rolling its own?